subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -608,6 +618,73 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
 }
 #endif /* defined(CONFIG_AWS_FOTA) */
 
+static int nct_settings_set(const char *key, size_t len_rd,
+				 settings_read_cb read_cb, void *cb_arg)
+{
+	if (!key) {
+		return -EINVAL;
+	}
+
+	int read_val;
+
+	LOG_DBG("Settings key: %s", key);
+
+	if (!strncmp(key, SETTINGS_KEY_PERSISTENT_SESSION,
+		     strlen(SETTINGS_KEY_PERSISTENT_SESSION))) {
+		if (read_cb(cb_arg, (void*)&read_val, len_rd) == len_rd) {
+			LOG_DBG("Read setting val: %d", read_val);
+#if defined(CONFIG_CLOUD_PERSISTENT_SESSIONS)
+			persistent_session = (bool)read_val;
+#endif
+			return 0;
+		}
+	}
+	return -ENOTSUP;
+}
+
+int save_session_state(const int session_valid)
+{
+	int ret = 0;
+
+#if defined(CONFIG_CLOUD_PERSISTENT_SESSIONS)
+	LOG_DBG("Setting session state: %d", session_valid);
+	persistent_session = (bool)session_valid;
+	ret = settings_save_one(SETTINGS_FULL_PERSISTENT_SESSION,
+				&session_valid, sizeof(session_valid));
+#endif
+	return ret;
+}
+
+static int nct_settings_init(void)
+{
+	int err = 0;
+
```

Comment:
The `if defined` can be moved up here to not need the `UNUSED_VARIABLE()` macro

Reply:
but then there would be an unused function warning for `nct_settings_set` 🙃

Reply:
Oh, that's true :(